src/coreclr/jit/emitriscv64.cpp

```diff
@@ -3503,51 +3513,76 @@ void emitter::emitDispInsName(
-            // if (imm12 & 0x800)
-            //{
-            //    imm12 |= 0xfffff000;
-            //}
+            unsigned opcode2      = (code >> 12) & 0x7;
+            unsigned rd           = (code >> 7) & 0x1f;
+            unsigned rs1          = (code >> 15) & 0x1f;
+            int      imm12        = static_cast<int>(code) >> 20;
+            bool     isHex        = false;
+            bool     hasImmediate = true;
+            int      printLength  = 0;
+
             switch (opcode2)
             {
-                case 0x0: // ADDI
-                    printf("addi           %s, %s, %d\n", rd, rs1, imm12);
-                    return;
-                case 0x1:                                                         // SLLI
-                    printf("slli           %s, %s, %d\n", rd, rs1, imm12 & 0x3f); // 6 BITS for SHAMT in RISCV64
-                    return;
-                case 0x2: // SLTI
-                    printf("slti           %s, %s, %d\n", rd, rs1, imm12);
-                    return;
-                case 0x3: // SLTIU
-                    printf("sltiu          %s, %s, %d\n", rd, rs1, imm12);
-                    return;
-                case 0x4: // XORI
-                    printf("xori           %s, %s, 0x%x\n", rd, rs1, imm12);
-                    return;
-                case 0x5: // SRLI & SRAI
-                    if (((code >> 30) & 0x1) == 0)
+                case 0x0: // ADDI & MV & NOP
+                    if (imm12 != 0)
+                    {
+                        printLength = printf("addi");
+                    }
+                    else if ((rd != REG_ZERO) || (rs1 != REG_ZERO))
                     {
-                        printf("srli           %s, %s, %d\n", rd, rs1, imm12 & 0x3f); // 6BITS for SHAMT in RISCV64
+                        printLength  = printf("mv");
+                        hasImmediate = false;
                     }
                     else
                     {
-                        printf("srai           %s, %s, %d\n", rd, rs1, imm12 & 0x3f); // 6BITS for SHAMT in RISCV64
+                        printf("nop\n");
+                        return;
                     }
-                    return;
+                    break;
+                case 0x1: // SLLI
+                    printLength = printf("slli");
+                    imm12 &= 0x3f; // 6 BITS for SHAMT in RISCV64
+                    break;
+                case 0x2: // SLTI
+                    printLength = printf("slti");
+                    break;
+                case 0x3: // SLTIU
+                    printLength = printf("sltiu");
+                    break;
+                case 0x4: // XORI
+                    printLength = printf("xori");
+                    isHex       = true;
+                    break;
+                case 0x5: // SRLI & SRAI
+                    printLength = printf((((code >> 30) & 0x1) == 0) ? "srli" : "srai");
```

Comment:
While here, our spec defines only 010000 and 000000. Please print `<unknown>` or sth like that if the remaining bits of the top 6 in imm12 are anything but 0. That's what llvm-objdump is doing.

Reply:
I've added additional logic to check whether this constant is valid, but I've used an illegal instruction format to keep it more consistent with the rest of disassm. Also added checks to the 32-bit version due to the fact those also have similar constants and fixed minor bug in there

Reply:
Ok will add for non-immediates too

Reply:
Done for other instructions